Packages/com.unity.inputsystem/InputSystem/Controls/InputControlPath.cs

```diff
@@ -721,14 +721,33 @@ public static bool Matches(string expected, InputControl control)
             return MatchesRecursive(ref parser, control);
         }
 
-        internal static bool MatchControlComponent(in ParsedPathComponent expectedControlComponent, ref InputControlLayout.ControlItem controlItem)
+        internal static bool MatchControlComponent(in ParsedPathComponent expectedControlComponent, ref InputControlLayout.ControlItem controlItem, bool matchAlias = false)
         {
-            // Match name.
-            if (!expectedControlComponent.m_Name.isEmpty)
+            // Check to see that there is a match with the name or alias if specified
+            // Exit early if we can't create a match.
+            if (!StringMatches(expectedControlComponent.m_Name, controlItem.name))
             {
-                if (!StringMatches(expectedControlComponent.m_Name, controlItem.name))
+                if (matchAlias)
+                {
+                    bool hasMatchingAlias = false;
+
+                    var aliases = controlItem.aliases;
+                    for (var i = 0; i < aliases.Count; i++)
+                    {
+                        if (StringMatches(expectedControlComponent.m_Name, aliases[i]))
+                        {
+                            hasMatchingAlias = true;
+                            break;
+                        }
+                    }
+
+                    if(!hasMatchingAlias)
```

Comment:
This would prevent it matching usage versions.


I think this might be an improved version that works


internal static bool MatchControlComponent(in ParsedPathComponent expectedControlComponent, ref InputControlLayout.ControlItem controlItem, bool matchAlias = false)
        {
            // Check to see that there is a match with the name or alias if specified
            // Exit early if we can't create a match.
            if (StringMatches(expectedControlComponent.m_Name, controlItem.name))
                return true;

            if (matchAlias)
            {
                var aliases = controlItem.aliases;
                for (var i = 0; i < aliases.Count; i++)
                {
                    if (StringMatches(expectedControlComponent.m_Name, aliases[i]))
                    {
                        return true;
                    }
                }
            }
                   

            // All of usages should match to the one of usage in the control
            foreach (var usage in expectedControlComponent.m_Usages)
            {
                if (!usage.isEmpty)
                {
                    var usageCount = controlItem.usages.Count;
                    for (var i = 0; i < usageCount; ++i)
                    {
                        if (StringMatches(usage, controlItem.usages[i]))
                        {
                            return true;
                        }
                    }
                }
            }

            return false;
        }

Reply:
<Gamepad>/{Back}

No longer matches with this update

Reply:
I think we need more test cases added to the test to verify what does and doesn't match.

Reply:
Added 2 new test cases to cover the Alias matching case and the Usages matching case.